web/src/narrow.js

```diff
@@ -199,6 +199,31 @@ export function activate(raw_terms, opts) {
             final_select_id: undefined,
         };
 
+        /*  (issue #14971)
+
+            Previously, using "first" or "latest" as final_select_id
+            caused errors due to data type mismatch.
+            This change employs conditional assignments:
+
+            * When then_select_id is "first", target_id is set to a special value (-1).
+              This sets final_select_id to assign "oldest" to the anchor, fetching
+              messages starting from the first message.
+
+            * Similarly, for "latest", target_id is set to another special value
+              (LARGER_THAN_MAX_MESSAGE_ID) to assign "newest" to the anchor,
+              fetching messages starting from the latest message.
+
+            NB: Review the switch statement below to see how the special values are used
+        */
+        const home_key = opts.trigger === "home" && opts.then_select_id === "first";
+        const end_key = opts.trigger === "end" && opts.then_select_id === "latest";
+        if (home_key) {
+            id_info.target_id = -1;
+        }
+        if (end_key) {
+            id_info.target_id = LARGER_THAN_MAX_MESSAGE_ID;
+        }
```

Comment:
Should we use something other than `then_select_id`? It doesn't seem to be used later down the code and our code expects it to be a number.

Reply:
@amanagr Thank you for your suggestion, Since it turns out `then_select_id` isn't directly used in `narrow.activate`, I'll remove it from the relevant section in `navigate.js` to maintain consistency and avoid potential confusion.
Please let me know if you have any further suggestions or require any additional information.